JavaScript/Angular.md

```diff
@@ -0,0 +1,1551 @@
+
+Install Angular CLI globally with `npm install -g @angular/cli` and check out what you can do with `ng help`.
+
+#### Creating a new project
+
+Use Angular CLI for initial project setup:
+
+```bash
+ng new my-app
+```
+
+As of version 7 there will be a quick wizard-like setup prompting you about routing and styling. Choose depending on your needs. All our Angular projects here at Infinum use routing and SCSS styling.
+
+#### Scaffolding
+
+You can use Angular CLI for generating new modules, components, services, directives, etc.
+
+Command is `ng generate [schematic-name]`, for example: `ng generate component`. There is also a shorthand for many generation commands, for example: `ng g c` === `ng generate component`. Run `ng generate --help` to see a list of all available options and schematics.
+
+##### Creating new modules
+
+To create a new module, use `ng g m MyModule`. If the module should have routes, you can generate module with routing skeleton with `ng g m MyModule --routing`.
+
+##### Creating new components
+
+Create components with `ng g c MyComponent` or `ng g c my-component`. In both cases the resulting component class name will be `MyComponent` and dir name will be `my-component`.
+
+Usually when creating components you want to create component module as well. This makes the component more modular and ensures that everything that the component needs is provided by the component module. Component module should declare and export the component - you can think of it as a public API. Component module could also declare some other "internal" components, but it does not have to necessarily export those internal components. It is our recommendation that component modules export only one declared component.
+
+Also remember that component can be declared only once (in one module), so it makes sense for reusable components to have their own module.
+
+Example:
+
+```bash
+# create a module
+ng g m Calendar
+
+# create the component - this will also automatically declare the component in the above created module
+ng g c Calendar
+
+# make sure to export CalendarComponent in CalendarModule
+
+cd calendar
+
+# add some "internal" components which are not exported
+ng g c WeekView
+ng g c DayView
+
+# WeekView and DayView do not have to be exported, as they are used only internally
+```
+
+```ts
+@NgModule({
+  imports: [
+    FormsModule,
+    MatButtonModule,
+  ],
+  declarations: [
+    CalendarComponent,
+    WeekViewComponent,
+    DayViewComponent,
+  ],
+  exports: [
+    CalendarComponent,
+  ]
+})
+export class CalendarModule { }
+```
+
+##### Creating new services
+
+Use `ng g s MyService` to generate new service. NOTE: unlike during component generation, service generation will not create a new dir for the service (although that might change in the future). We recommend that you first create a dir for your service and then generate new service inside that dir.
```

Comment:
I believe that using, for example: `ng g s services/my-service/my-service` also works without the extra step of a manual folder creation